osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -417,11 +421,14 @@ private async Task clearDatabaseScores(MultiplayerRoom room)
 
         private async Task updateDatabaseSettings(MultiplayerRoom room)
         {
+            if (room.Settings.RulesetID < 0 || room.Settings.RulesetID > ILegacyRuleset.MAX_LEGACY_RULESET_ID)
+                throw new InvalidStateException("Attempted to select a beatmap of a custom ruleset.");
```

Comment:
```suggestion
                throw new InvalidStateException("Attempted to select an unsupported ruleset.");
```

It has nothing to do with the beatmap (the setting is at a room level and may be a convert).